pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -283,32 +347,89 @@ else let
   references = nativeBuildInputs ++ buildInputs
             ++ propagatedNativeBuildInputs ++ propagatedBuildInputs;
 
+  getSpliced = type: dep:
+    dep.__spliced or
+      (let approximate-pname = dep.pname or dep.name; in
+        assert
+          (dep?stdenv &&
+           dep.stdenv.buildPlatform != dep.stdenv.targetPlatform &&
+
+           # I can't test on Darwin, so let's just ignore it for now.
+           !dep.stdenv.buildPlatform.isDarwin &&
+           !dep.stdenv.hostPlatform.isDarwin &&
+           !dep.stdenv.targetPlatform.isDarwin &&
+
+           !(list-of-shame ? ${approximate-pname})
+          )
+
+            # Hi there!  If you're reading this, it's probably
```

Comment:
This would be way easier to read and a lot less lines if we'd wrote one setence per line.

Reply:
Yes it would be but that's part of #263082. I feel like if that got merged it would force us to address the cross-compilation issues.